node/src/components/consensus/highway_core/state/tests.rs

```diff
@@ -252,6 +252,48 @@ fn fork_choice() -> Result<(), AddVoteError<TestContext>> {
     Ok(())
 }
 
+#[test]
+fn round_exp_validity() -> Result<(), AddVoteError<TestContext>> {
+    let params = Params::new(
+        0, // seed
+        TEST_BLOCK_REWARD,
+        TEST_BLOCK_REWARD / 5,
+        TEST_REWARD_DELAY,
+        2, // minimum round exponent
+        u64::MAX,
+        Timestamp::from(u64::MAX),
+    )
+    .with_init_round_exp(6);
+    let mut state = State::new(&[Weight(1), Weight(2), Weight(1)], params);
+    let mut rng = TestRng::new();
+
+    // The initial round exponent is 6, so Alice and Carol count as having round exponent 6.
+    // Bob starts with 7. Now 6 and 7 are both median values.
+    let b0 = add_vote!(state, rng, BOB, 0, 7u8, 0xB0; N, N, N)?;
+    // Alice is allowed to start with 5 and even go to 4, which is only two below the median 6.
+    let a0 = add_vote!(state, rng, ALICE, 1, 5u8, None; N, b0, N)?;
+    let a1 = add_vote!(state, rng, ALICE, 32 + 21, 4u8, None; a0, b0, N)?;
```

Comment:
What's the `32+21`?

Reply:
Two-thirds through the second round. When Alice is supposed to create a witness vote. (Although we don't check that exact timing.)